File: Data Structures & Algorithms/longest-turbulent-subarray/submission-2.py
class Solution:
    def maxTurbulenceSize(self, arr: list[int]) -> int:
        if not arr: 
            return 0
        n = len(arr)
        if n == 1: 
            return 1
        
        max_len = 1
        left = 0
        
        for right in range(1, n):
            # 获取当前比较结果：1 (>) , -1 (<), 0 (=)
            c = (arr[right-1] > arr[right]) - (arr[right-1] < arr[right])
            
            if c == 0:
                left = right
            elif right == n - 1 or c * ((arr[right] > arr[right+1]) - (arr[right] < arr[right+1])) != -1:
                max_len = max(max_len, right - left + 1)
                left = right
                
        return max_len

File: Data Structures & Algorithms/sort-colors/submission-2.go
func sortColors(nums []int) {
    left, i := 0, 0
    right := len(nums) - 1

    for i <= right {
        if nums[i] == 0 {
            // 遇到 0，交换到左侧
            nums[left], nums[i] = nums[i], nums[left]
            left++
            i++
        } else if nums[i] == 2 {
            // 遇到 2，交换到右侧
            nums[right], nums[i] = nums[i], nums[right]
            right--
            // 注意：此时 i 不增加，因为交换过来的数需要再次判断
        } else {
            // 遇到 1，直接跳过
            i++
        }
    }
}